### PR TITLE
add rules agreement page

### PR DESCRIPTION
**Pull request by mkwia**
[preview](http://dev.mkwia.com/lainchan-rules-popup/)

fixed issues with the [initial pull request](https://github.com/lainchan/lainchan/pull/55)

### How to use:
- make sure `$config['additional_javascript'][] = 'js/jquery.min.js';` is in _instance-config.php_
- add `$config['additional_javascript'][] = 'js/rules-popup.js';` directly after the aforementioned line in _instance-config.php_

### CODE REVIEW

File: js/rules-popup.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * 2016 mkwia <github.com/mkwia>
+ *
+ * Usage:
+ *   $config['additional_javascript'][] = 'js/jquery.min.js';
+ *   $config['additional_javascript'][] = 'js/rules-popup.js';
+ *
+ */
+
+$(function() {
+    if (typeof localStorage.rulesAccepted === "undefined") + function(c) {
```
Comment (barrucadu) [inline js/rules-popup.js]: I'm not really a javascript person, what does the `+` here mean? Wouldn't it be clearer to do:

```js
if (typeof localStorage.rulesAccepted === "undefined") {
    var captcha = /* generate a random string */
    /* the rest of the stuff, with no additional function call */
}
```

File: js/rules-popup.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "<input type='submit' value='ACCEPT' />" +
+            "</form>" +
+            "</div>" +
+            "</div>" +
+
+            "");
+
+        $("#rules-actual").load("/templates/rules.html");
+
+    }("faggotry1234");
```
Comment (barrucadu) [inline js/rules-popup.js]: I think this should be a random string.

File: js/rules-popup.js (hunk 1 above)
Comment (mkwia) [inline js/rules-popup.js]: Some JQuery thing; not really needed.

**Comment by barrucadu** [comment]
Could you move all the css to style.css? Defining it all in the javascript/html like that makes it awkward to override in specific themes.

**Comment by mkwia** [comment]
Are you going to want to override the rules popup depending on the theme? To me it seems like that should be kept uniform.

**Comment by barrucadu** [comment]
Probably not to a large degree, but colours will probably get changed.

File: js/rules-popup.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    $("#rules-popup")
+        .append("<div class='rules-popup-top'>lainchan rule agreement</div>")
+        .append("<div class='rules-popup-content-wrapper'></div>")
+        .append("<div class='rules-popup-bottom'></div>");
+
+    $(".rules-popup-content-wrapper")
+        .append("<div id='rules-popup-content'></div>");
+    $("#rules-popup-content")
+        .load("rules.html");
```
Comment (barrucadu) [inline js/rules-popup.js]: Shouldn't this be `/templates/rules.html`? `lainchan.org/rules.html` is where the generated page with the rest of the HTML template will go.

**Pull request merged**